From 2fcd97831febcb6d61c354cb6d0e62088c51df23 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Thu, 2 Jun 2016 13:30:14 +0200
Subject: [PATCH 0518/2241] mm, page_alloc: fix memblock reserve

Fix regression that was introduced by commit 7e18adb4f80b ("mm: meminit:
initialise remaining struct pages in parallel with kswapd"), which in
fact disabled memblock reserve at all for all platforms not using
CONFIG_DEFERRED_STRUCT_PAGE_INIT (x86 is the only user), hence
temporary shortage of memory possible to allocate during tests.

Fix 4/4 of SYSTEMSW-2443: "Armada 3700 page alloc failures"

Change-Id: Ifead508fdda4aaadffc15d012ae499813e82819b
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30482
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Lior Amsalem <alior@marvell.com>
Reviewed-by: Wilson Ding <dingwei@marvell.com>
Verified-Armada8K: Nadav Haklai <nadavh@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Verified-Armada37x0: Nadav Haklai <nadavh@marvell.com>
Verified-Armada7K: Nadav Haklai <nadavh@marvell.com>
---
 mm/page_alloc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/page_alloc.c b/mm/page_alloc.c
index 760c1bd..67ad523 100644
--- a/mm/page_alloc.c
+++ b/mm/page_alloc.c
@@ -325,7 +325,7 @@ static inline bool early_page_uninitialised(unsigned long pfn)
 
 static inline bool early_page_nid_uninitialised(unsigned long pfn, int nid)
 {
-	return false;
+	return true;
 }
 
 static inline bool update_defer_init(pg_data_t *pgdat,
-- 
2.7.4

